toucan_connectors/databricks/databricks_connector.py

```diff
@@ -0,0 +1,129 @@
+    connect_timeout: int = None
+
+    @backoff.on_exception(backoff.expo, pyodbc.Error, max_time=CLUSTER_START_TIMEOUT)
+    def _connect(self) -> Any:
+        """
+        In case of on-demand clusters, the connector might fail to connect.
+        However, the first connection trial starts the cluster so this method will exponentially retry
+        to connect until CLUSTER_START_TIMEOUT is reached.
+        """
+        try:
+            return pyodbc.connect(self._build_connection_string(), autocommit=True, ansi=self.Ansi)
+        except pyodbc.InterfaceError as e:
+            if len(e.args) >= 2 and 'Authentication/authorization error occured' in e.args[1]:
+                raise DataBricksUnauthorizedError('Invalid credentials')
+            else:
+                raise DataBricksConnectionError('Invalid connection params')
+
+    def _build_connection_string(self) -> str:
+        """For constants see: https://docs.databricks.com/integrations/bi/jdbc-odbc-bi.html"""
+        connection_params = {
+            'Driver': '/opt/simba/spark/lib/64/libsparkodbc_sb64.so',  # This path must be correct when installing
+            # ODBC driver
+            'Host': '127.0.0.0.1' if self.Host == 'localhost' else self.Host,
+            'Port': self.Port,
+            'HTTPPath': self.HTTPPath,
+            'ThriftTransport': 2,
+            'SSL': 1,
+            'AuthMech': 3,
+            'UID': 'token',
+            'PWD': self.PWD.get_secret_value(),
+            'connect_timeout': self.connect_timeout,
+        }
+        return ';'.join(f'{k}={v}' for k, v in connection_params.items() if v is not None)
+
+    @staticmethod
+    def _get_details(index: int, status: Optional[bool]):
+        checks = ['Host resolved', 'Port opened', 'Connected to Databricks', 'Authenticated']
+        ok_checks = [(c, True) for i, c in enumerate(checks) if i < index]
+        new_check = (checks[index], status)
+        not_validated_checks = [(c, None) for i, c in enumerate(checks) if i > index]
+        return ok_checks + [new_check] + not_validated_checks
+
+    def get_status(self) -> ConnectorStatus:
+        # Check hostname
+        try:
+            self.check_hostname(self.Host)
+        except Exception as e:
+            return ConnectorStatus(status=False, details=self._get_details(0, False), error=str(e))
+        # Check port
+        try:
+            self.check_port(self.Host, self.Port)
+        except Exception as e:
+            return ConnectorStatus(status=False, details=self._get_details(1, False), error=str(e))
+        try:
+            self._connect()
+        except DataBricksConnectionError as e:
+            return ConnectorStatus(
+                status=False, details=self._get_details(2, False), error=e.args[0]
+            )
+
+        except (DataBricksUnauthorizedError, Exception) as e:
+            return ConnectorStatus(
+                status=False, details=self._get_details(3, False), error=e.args[0]
+            )
```

Comment:
IMO these Exceptions catch should be directly inside the  `_connect` method, what do you think ?

Reply:
No, I don't think so, we specifically raise them in `_connect` to provide the correct details in `get_status`